ext/phar/zip.c

```diff
@@ -683,6 +681,8 @@ int phar_parse_zipfile(php_stream *fp, char *fname, size_t fname_len, char *alia
 		return FAILURE;
 	}
 
+	mydata->fp = fp;
```

Comment:
Looks like there's a bunch of usages above that use PHAR_ZIP_FAIL rather than phar_destroy_phar_data -- just want to double check whether using phar_destroy_phar_data is really okay here.

Reply:
Ah, right! At least for consistency, we should use the `PHAR_ZIP_FAIL()` macro. However, not calling `phar_destroy_phar_data()` keeps some memory allocated until request shutdown; that is handled differently in ext/phar/tar.c. Might be worthwhile to give this a closer review.